spyderlib/widgets/editor.py

```diff
@@ -64,106 +161,175 @@ def __init__(self, parent, tabs, fullpath_sorting):
         # (e.g. the editor's analysis thread in Spyder), thus leading to
         # a segmentation fault on UNIX or an application crash on Windows
         self.setAttribute(Qt.WA_DeleteOnClose)
-
-        self.indexes = None
-
-        self.setWindowIcon(get_icon('filelist.png'))
-        self.setWindowTitle(_("File list management"))
-
+        
+        self.setWindowFlags(Qt.Dialog | Qt.FramelessWindowHint) 
+        self.setWindowOpacity(0.9)
         self.setModal(True)
 
-        flabel = QLabel(_("Filter:"))
+        self.tabs = tabs
+        self.tab_data = tab_data
+        self.filtered_index_to_path = None
+        self.full_index_to_path = None
+        self.path_to_line_count = None
+        
         self.edit = QLineEdit(self)
-        self.edit.returnPressed.connect(self.edit_file)
-        self.edit.textChanged.connect(lambda text: self.synchronize(0))
-        fhint = QLabel(_("(press <b>Enter</b> to edit file)"))
+        self.up_down_filter = UpDownFilter()
+        self.up_down_filter.up_down.connect(self.handle_up_down)
+        self.edit.installEventFilter(self.up_down_filter)
+        self.edit.returnPressed.connect(self.handle_edit_file)
+        self.edit.textChanged.connect(lambda text: self.synchronize(None))
+        new_shortcut("Ctrl+W", self, lambda: self.close_file.emit(self.filtered_index_to_full(self.listwidget.currentRow())))

```

Comment:
I'm confused.  Esc closes the switcher, and takes you back to where you were. Ctrl-W closes whichever tab you had selected in the switcher, but leaves the switcher open (unless there are no more tabs in which case it closes on the next ``synchronize``).